### PR TITLE
Kuz-548 implements kuzzle-common-objects

### PR DESCRIPTION
**Pull request by jenow**
- Removed core/models/requestObject, core/models/responseObject and core/models/errors
- Implemented kuzzle-common-objects wich contains requestObject, responseObject and all errors

### CODE REVIEW

File: test/services/implementations/elasticsearch.test.js [hunk 1]
```diff
@@ -5,9 +5,9 @@ var
   rewire = require('rewire'),
   params = require('rc')('kuzzle'),
   Kuzzle = require.main.require('lib/api/Kuzzle'),
-  RequestObject = require.main.require('lib/api/core/models/requestObject'),
-  BadRequestError = require.main.require('lib/api/core/errors/badRequestError.js'),
-  NotFoundError = require.main.require('lib/api/core/errors/notFoundError'),
+  RequestObject = require.main.require('kuzzle-common-objects').Models.requestObject,
+  BadRequestError = require.main.require('kuzzle-common-objects').Errors.badRequestError.js,
```
Comment (stafyniaksacha) [inline test/services/implementations/elasticsearch.test.js]: use `badRequestError` instead of `badRequestError.js`

**Comment by stafyniaksacha** [comment]
:+1:

File: package.json [hunk 2]
```diff
@@ -40,6 +40,7 @@
     "ioredis": "^1.15.1",
     "json-stable-stringify": "1.0.1",
     "jsonwebtoken": "^5.4.0",
+    "kuzzle-common-objects": "git+https://github.com/kuzzleio/kuzzle-common-objects.git",
```
Comment (ballinette) [inline package.json]: Should we fix a tagged version of the kuzzle-common-objects source ?

**Comment by dbengsch** [comment]
+1

**Pull request merged**